src/tests/facebookOauth.test.js

```diff
@@ -0,0 +1,62 @@
+import chai from 'chai';
+import http from 'chai-http';
+import app from '../app';
+import { mock } from "sinon";
+import 'dotenv/config';
+import {facebookFunction , user} from  '../controllers/facebookOauth'
+
+chai.use(http);
+const { expect } = chai;
+
+const mockUser = {
+    id: 1,
+    firstName: "test@gmail.com",
+    lastName: "test2",
+    roleId:5
+  };
+
+describe("LOGIN WITH FACEBOOK",() =>{
+
+    it('should user login with facebook account', async() =>{
+
+        const mockProvider = 'facebook';
+        const res = await chai.request(app).get(`/auth/${mockProvider}`);
+
+      expect(res.status).to.equal(200);
+
+        
+      
+    });
+    it('should user login with facebook callback', async() =>{
+
+        const mockProvider = 'facebook';
+        const res = await chai.request(app).get(`/auth/${mockProvider}/callback`);  
+      expect(res.status).to.equal(200);        
+    });
+   
+      it("should call done once after find or create user", async () => {
+        const doneMock = mock();
+        const mockUser = {
+          _json: {
+            first_name: "testuser1",
+            last_name: "testuser",
+          }
+        };
+  
+        await facebookFunction(null, null, mockUser, doneMock);
```

Comment:
What are you testing here? What are you expecting?

Reply:
I am testing the function itself